zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteManager.java

```diff
@@ -417,6 +417,11 @@ private boolean isNotePathAvailable(String notePath) {
     return true;
   }
 
+  public synchronized String getNoteIdByPath(String notePath) throws IOException {
```

Comment:
why add `synchronized`?

Reply:
> why add `synchronized`?

Thanks @zjffdu  for reviews.  it's been removed.